src/tools/sfdk/sdkmanager.cpp

```diff
@@ -1348,15 +1348,32 @@ bool SdkManager::mapEnginePaths(QString *program, QStringList *arguments, QStrin
         {cleanSharedSrc, Constants::BUILD_ENGINE_SHARED_SRC_MOUNT_POINT, caseInsensitiveOnWindows}
 #endif
     };
+    *program = QDir::fromNativeSeparators(*program);
+
     for (const Mapping &mapping : mappings) {
         qCDebug(sfdk) << "Mapping" << mapping.hostPath << "as" << mapping.enginePath;
         program->replace(mapping.hostPath, mapping.enginePath, mapping.cs);
-        arguments->replaceInStrings(mapping.hostPath, mapping.enginePath, mapping.cs);
+        QRegularExpression::PatternOption reCaseSensitivity = (mapping.cs == Qt::CaseInsensitive)
+                ? QRegularExpression::CaseInsensitiveOption : QRegularExpression::NoPatternOption;
+        QRegularExpression re(QString("(?:%1|%2)([^:;,'\"]*)")
+                              .arg(QRegularExpression::escape(mapping.hostPath))
+                              .arg(QRegularExpression::escape(QDir::toNativeSeparators(mapping.hostPath))),
+                              reCaseSensitivity);
+        auto map = [=](QString string) {
+            QRegularExpressionMatchIterator i = re.globalMatch(string);
+            while (i.hasNext()) {
+                QRegularExpressionMatch match = i.next();
+                QString fullNativeHostPath = match.captured(0);
+                QString fullEnginePath = QDir::fromNativeSeparators(mapping.enginePath + match.captured(2));
+                string.replace(fullNativeHostPath, fullEnginePath);
+            }
```

Comment:
It would still not work e.g. with `"c:\users\user\foo;c:\users\user\foo\bar\baz"` I think. We need to sort matches by length and replace longer matches first. Maybe build `QMap<QString, QString>` first and backward-iterate that doing the replacements.

The capture number should be 1 instead of 2, the first group is non capturing.

Reply:
> The capture number should be 1 instead of 2, the first group is non capturing.

These write-only languages are hard :)